tests/test_jwt_utils.py

```diff
@@ -30,19 +31,17 @@ def test_get_matching_missing_key(self) -> None:
 
 
 class TestDecodeJWT:
-    @patch("lbz.jwt_utils.get_matching_jwk", return_value={})
-    def test_did_not_find_matching_jwk(
-        self, get_matching_jwk_mock: MagicMock, caplog: pytest.LogCaptureFixture
-    ) -> None:
-        with pytest.raises(Unauthorized):
-            decode_jwt("x")
-        get_matching_jwk_mock.assert_called_once_with("x")
-        assert "Failed decoding JWT with following details" in caplog.text
-
-    @patch("lbz.jwt_utils.get_matching_jwk", return_value={})
+    def test_did_not_find_matching_jwk(self, caplog: pytest.LogCaptureFixture) -> None:
+        public_key = deepcopy(SAMPLE_PUBLIC_KEY)
+        public_key["kid"] = "x"
```

Comment:
Also, you should return an empty list - it will be more obvious instead of overwriting one key 😉

Reply:
That would mean that you can simply use a decorator as it was done before.